### PR TITLE
Log incidents diff during route refresh

### PR DESCRIPTION
**Pull request by vadzim-vys**
### Description
This is a follow up PR that adds logging of incidents diff: https://github.com/mapbox/mapbox-navigation-android/pull/5767#discussion_r884942577

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt [hunk 1]
```diff
@@ -15,16 +16,30 @@ internal class DirectionsRouteDiffProvider {
         val newRouteLegs = newRoute.directionsRoute.legs() ?: return emptyList()
         val routeDiffs = arrayListOf<String>()
         for (legIndex in currentLegIndex until min(oldRouteLegs.size, newRouteLegs.size)) {
-            val oldLegAnnotation = oldRouteLegs[legIndex].annotation()
-            val newLegAnnotation = newRouteLegs[legIndex].annotation()
-            val updatedAnnotations = getUpdatedAnnotations(oldLegAnnotation, newLegAnnotation)
+            val oldLeg = oldRouteLegs[legIndex]
```
Comment (RingerJK) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt:19]: add please route index, as we'll use the same for alternatives

**Review by RingerJK — APPROVED** [review]
LGTM with 1 comment

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt (hunk 1 above)
Comment (VysotskiVadim) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt:19]: great point. Do you mind if I add it during implementation of route alternatives refresh. I don't exactly know how I will be using routes diff provider.

Comment (RingerJK) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/DirectionsRouteDiffProvider.kt:19]: sure, let's then mention it in the original issue

**Pull request merged**